lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1072,35 +1174,86 @@ def install(collection, path, artifacts_manager):  # FIXME: mv to dataclasses?
         u"Installing '{coll!s}' to '{path!s}'".
         format(coll=to_text(collection), path=collection_path),
     )
+    metadata_path = os.path.join(path, f"{collection.namespace}.{collection.name}-{collection.ver}.info")
+    b_metadata_path = to_bytes(metadata_path, errors='surrogate_or_strict')
 
     if os.path.exists(b_collection_path):
         shutil.rmtree(b_collection_path)
+    if os.path.exists(b_metadata_path):
+        shutil.rmtree(b_metadata_path)
 
     if collection.is_dir:
         install_src(collection, b_artifact_path, b_collection_path, artifacts_manager)
     else:
-        install_artifact(b_artifact_path, b_collection_path, artifacts_manager._b_working_directory)
+        install_artifact(
+            b_artifact_path,
+            b_collection_path,
+            artifacts_manager._b_working_directory,
+            collection.signatures,
+            artifacts_manager.keyring
+        )
+        write_source_metadata(
+            collection,
+            b_metadata_path,
+            artifacts_manager
+        )
 
     display.display(
         '{coll!s} was installed successfully'.
         format(coll=to_text(collection)),
     )
 
 
-def install_artifact(b_coll_targz_path, b_collection_path, b_temp_path):
+def write_source_metadata(collection, b_metadata_path, artifacts_manager):
+    # type: (Candidate, bytes, ConcreteArtifactsManager) -> None
+    source_data = artifacts_manager.get_galaxy_artifact_source_info(collection)
+    if source_data.get("signatures") is None:
+        return
+    b_yaml_source_data = to_bytes(yaml_dump(source_data), errors='surrogate_or_strict')
+    dest_file = os.path.join(b_metadata_path, to_bytes(SOURCE_METADATA_FILE))
+    try:
+        os.mkdir(b_metadata_path, mode=0o0755)
+        with open(dest_file, mode='w+b') as fd:
+            os.chmod(fd.name, 0o0644)
+            fd.write(b_yaml_source_data)
+    except Exception:
```

Comment:
How about the rest of the possible exceptions?
```suggestion
    except BaseException:
```

Reply:
Isn't that risky, you could end up accidentally masking bad code. We don't intercept BaseException anywhere else in ansible-galaxy. I was actually wondering if this (and maybe the same pattern install_artifact) should be more specific and just catch OSError, since that's the expected potential error.